SC_redesign/Crop_and_Soil/crop/crop_management.py

```diff
@@ -0,0 +1,204 @@
+        if self.data.harvest_index <= 1.0:
+            self.data.cut_biomass = self.data.above_ground_biomass * self.data.harvest_index  # SWAT 5:2.4.2
+        else:
+            self.data.cut_biomass = self._determine_biomass_cut_from_whole_plant(self.data.biomass,
+                                                                                 self.data.harvest_index)
+        fraction_cut = self.data.cut_biomass / self.data.biomass
+        self.data.biomass -= self.data.cut_biomass
+
+        # Reset some growth parameters (SWAT 6:1.2)
+        self.data.leaf_area_index = self.data.leaf_area_index * (1 - fraction_cut)
+        self.data.accumulated_heat_units = self.data.accumulated_heat_units * (1 - fraction_cut)
+
+        # Biomass collected as yield, and its nutrient content
+        self.data.yield_collected = self.data.cut_biomass * collected_fraction  # SWAT 5:3.3.4
+        if self.data.do_harvest_index_override:
+            self.data.yield_nitrogen = self.data.optimal_nitrogen_fraction * self.data.yield_collected  # SWAT 5:2.4.7
+            self.data.yield_phosphorus = self.data.optimal_phosphorus_fraction * self.data.yield_collected  # SWAT 5:2.4.8
+        else:
+            self.data.yield_nitrogen = self.data.yield_nitrogen_fraction * self.data.yield_collected  # SWAT 5:2.4.5
+            self.data.yield_phosphorus = self.data.yield_phosphorus_fraction * self.data.yield_collected  # SWAT 5:2.4.6
+
+        # Uncollected biomass and nutrients
+        self.data.yield_residue = self.data.cut_biomass * (1 - collected_fraction)  # SWAT 5:3.3.5
+        if self.data.do_harvest_index_override:
+            self.data.residue_nitrogen = self.data.optimal_nitrogen_fraction * self.data.yield_residue
+            self.data.residue_phosphorus = self.data.optimal_phosphorus_fraction * self.data.yield_residue
+        else:
+            self.data.residue_nitrogen = self.data.yield_nitrogen_fraction * self.data.yield_residue
+            self.data.residue_phosphorus = self.data.yield_phosphorus_fraction * self.data.yield_residue
+
+        # TODO: residue needs to be accumulated in the soil (Soil class) - GitHub Issue #245
+
+        # TODO: are above- and below-ground lignin residue (percent) needed?
+        #   in the old version, they were both hard-coded to 17 - GitHub Issue #163
+
+    # ---- Helper Methods ----
+    @staticmethod
+    def _determine_potential_harvest_index(heat_fraction: float, optimal_harvest_index: float) -> float:
+        """calculates the potential harvest index for a plant on a given day
+
+        Args:
+            heat_fraction: fraction of potential heat units accumulated to date
+            optimal_harvest_index: species-specific optimal harvest index for the plant at maturity under ideal
+                conditions
+
+        SWAT Reference: 5:2.4.1
+
+        Details: Harvest Index is the ratio of grain to total shoot dry matter
+
+        Returns: potential harvest index for the day
```

Comment:
```suggestion
        Returns: potential harvest index for the day (unitless)
```
I think this is unitless.